### ISSUE TITLE
Registered scripts and styles for use with acf blocks are loaded on all pages, not just on pages where block is included

### ISSUE THREAD

**Issue by Josh979** [AdvancedCustomFields/acf#770]
Am I doing something wrong? I wouldn't imagine this is the intended behavior, as it defeats the purpose of having components.

Steps to reproduce are as follows:

1. register a new ACF block using new block.json method with `register_block_type`.
2. register the styles and scripts to be used with the block by assigning them a handle via `wp_register_style` and `wp_register_script` functions for later use.
3. utilize the `style` and `script` properties of block.json in new ACF block to call those asset handles.
4. block's css and js files are now getting loaded on all pages across the site, regardless if a block is used on that page?

**Comment by CreativeDive** [comment]
@Josh979 I think this has more to do with WP than ACF.

Do you have tried the usage of `should_load_separate_core_block_assets` filter?

`add_filter( 'should_load_separate_core_block_assets', '__return_true' );`

**Comment by lgladdy** [comment]
Hey @Josh979,

This is indeed WordPress's intended behaviour. We don't have much choice but to use the native provided block script registration system with the upcoming move to iframed blocks, so we can't do much about this.

As CreativeDive mentioned above, that filter will move your block scripts to the footer, after WordPress knows which blocks have actually been used, so only includes the required ones.

**Comment by Josh979** [comment]
@CreativeDive Thank you - this is helpful. I am now able to only load on pages where the block is used. Is there a native way to have block styles/CSS load into the header with this filter active, instead of the footer? 

@lgladdy - thanks for the context. It's extremely odd that this is the default WP behavior!

**Comment by CreativeDive** [comment]
@Josh979 It seems that it depends on the type of theme (block theme / classic theme). In a block theme it should be possible to load the style in the head of the page: [https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/](https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/)

**Comment by FalkoJoseph** [comment]
I don't think you need to use `wp_register_style` and `wp_register_script` for blocks. I only use `script` and `style` in block.json which load the assets only when the block is used.

**Comment by Josh979** [comment]
@FalkoJoseph - Like @CreativeDive mentioned above, I think that it works differently depending on your theme type. I have a classic theme, and so the blocks load everywhere. I believe if I was instead utilizing a block theme, this would likely not be the case.
